securitas/tests/unit/controller/test_password.py

```diff
@@ -50,6 +112,48 @@ def test_password(client, dummy_user):
     )
     assert result.status_code == 200
     page = BeautifulSoup(result.data, 'html.parser')
+    print(page.prettify())
+    password_input = page.select("input[name='current_password']")[0]
+    assert 'is-invalid' in password_input['class']
+    invalidfeedback = password_input.find_next('div', class_='invalid-feedback')
+    assert (
+        invalidfeedback.get_text(strip=True)
+        == "The old password or username is not correct"
+    )
+
+
+def test_non_matching_passwords(client):
+    """Verify that passwords that dont match are caught"""
+    result = client.post(
+        '/password-reset?username=jbloggs',
+        data={
+            "current_password": "sdsds",
+            "password": "password1",
+            "password_confirm": "password2",
+        },
+    )
+    assert result.status_code == 200
+    page = BeautifulSoup(result.data, 'html.parser')
+    password_input = page.select("input[name='password']")[0]
+    assert 'is-invalid' in password_input['class']
+    invalidfeedback = password_input.find_next('div', class_='invalid-feedback')
+    assert invalidfeedback.get_text(strip=True) == "Passwords must match"
+
+
+@pytest.mark.vcr()
+def test_password(client, dummy_user):
```

Comment:
Could you also add a test like this one but without the `dummy_user` fixture, so that we test the case where the username does not exist? I don't think we need to display a different user message but I'd prefer it tested.

Reply:
done!